### ISSUE TITLE
Bug when assign function with annotation to class attribute

### ISSUE THREAD

**Issue by STofone** [microsoft/pyright#11271]
## Environment data

-   Pylance version: 2025.10.4
-   OS and version: win11
-   Python version (& distribution if applicable, e.g. Anaconda): 313

## Code Snippet
```python
def func1(a, b):
    print(a)
    print(b)


def func2(a: int, b):
    print(a)
    print(b)


class func_set:
    func1 = func1
    func2 = func2


func_set.func1(1,2)
func_set.func2(1,2)
```
func2 got error as below
```
Cannot access attribute "func2" for class "type[func_set]"
  Could not bind method "func2" because "func_set" is not assignable to parameter "a"
    "func_set" is not assignable to "int"Pylance[reportAttributeAccessIssue](https://github.com/microsoft/pylance-release/blob/main/docs/diagnostics/reportAttributeAccessIssue.md)
```

**Comment by erictraut** [comment]
Pyright's behavior here is by design, so I don't consider this a bug.

You're assigning an instance method here. When accessing an instance method through a class, the first argument is assumed to be a `self` parameter, and pyright validates that the class that you're binding to it is type compatible. The reason `func1` doesn't generate the same error is that the first parameter in that function is unannotated, so its type is inferred to be `Unknown`.

If you want your code to be sound and type safe, the attributes `func1` and `func2` should be assigned static methods.

```python
class func_set:
    func1 = staticmethod(func1)
    func2 = staticmethod(func2)
```
